docs/api.rst

```diff
@@ -977,6 +977,132 @@ Components
             "web_url": "http://example.com/projects/hello/weblate/"
         }
 
+.. http:patch:: /api/components/(string:project)/(string:component)/
+
+    Edit a component by a patch request.
+
+    :param project: Project URL slug
+    :type project: string
+    :param component: Component URL slug
+    :type component: string
+    :<json string name: name of component
+    :<json string slug: slug of component
+    :<json string repo: VCS repository URL
+    :>json string branch: VCS repository branch
+    :>json string file_format: file format of translations
+    :>json string filemask: mask of translation files in the repository
+    :>json string git_export: URL of the exported VCS repository with translations
+    :>json string license: license for translations
+    :>json string license_url: URL of license for translations
+    :>json string name: name of component
+    :>json string slug: slug of component
+    :>json object project: the translation project; see :http:get:`/api/projects/(string:project)/`
+    :>json string repo: VCS repository URL
+    :>json string template: base file for monolingual translations
+    :>json string new_base: base file for adding new translations
+    :>json string vcs: version control system
+    :>json string repository_url: URL to repository status; see :http:get:`/api/components/(string:project)/(string:component)/repository/`
+    :>json string translations_url: URL to translations list; see :http:get:`/api/components/(string:project)/(string:component)/translations/`
+    :>json string lock_url: URL to lock status; see :http:get:`/api/components/(string:project)/(string:component)/lock/`
+    :>json string changes_list_url: URL to changes list; see :http:get:`/api/components/(string:project)/(string:component)/changes/`
+    :>json string push: URL of a push repository
```

Comment:
Many of the fields are read only or links to other API endpoints, at least following do not make sense to update: license_url, repository_url, repository_url, lock_url, changes_list_url

Same applies to PATCH below.

Reply:
These are the outputs, not the input. :thinking:

Reply:
Ah, I didn't realize that. Maybe it would be useful to document inputs here and reference to GET for the outputs?

Reply:
Yeah. Makes sense.